scripts/generate-rate-limit-schema.ts

```diff
@@ -0,0 +1,18 @@
+import { writeFileSync } from 'node:fs';
+import zodToJsonSchema from 'zod-to-json-schema';
+import { rateLimiterConfigurationSchema } from '../src/server/lib/rate-limit/configurationValidator';
+
+process.stdout.write('✨  Generating rate limit configuration schema  ✨\n');
+
+// Generate a JSON schema for the rate limiter configuration.
+const jsonSchema = zodToJsonSchema(
+  rateLimiterConfigurationSchema,
+  'rateLimiterConfigurationSchema',
+);
+
+// Write the pretty printed JSON schema to .ratelimit.schema.json.
+// We use this for autocompletion when editing the rate limit configuration file (and example).
+writeFileSync(
+  '.ratelimit.schema.json',
+  JSON.stringify(jsonSchema, null, 2) + '\n',
+);
```

Comment:
there isn't any typescript specific stuff going on here, could this just be a `.js` file and we just use `node` rather than having to introduce `ts-node` as a dependency

i know `ts-node` is good, but maybe it's a bit overkill for this use case.

Reply:
if we were able to introduce ts-node to a wider part of this project, then i'd be all over it 😅

Reply:
i added `ts-node` because i'm importing `configurationValidator.ts` which has multiple typescript dependencies, couldn't think of a cleaner way to do it

Reply:
as it's just a dev dependency, it doesn't seem too bad to add it in - what d'you think?

Reply:
Yeah in agreement now, and we can move some of our other `js` scripts to `ts` by using `ts-node` :)